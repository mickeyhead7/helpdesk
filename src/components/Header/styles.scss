.header {
  $tabletUp: "only screen and (min-width: 40.0625em)";
  $desktopUp: "only screen and (min-width: 60.0625em)";

  background-color: rebeccapurple;
  margin: 0 0 1em;

  .headerContainer {
    align-items: center;
    margin: 0 auto;
    max-width: 80em;
    padding: 1em;

    @media #{$tabletUp} {
      display: flex;
      padding: 1em;
    }

    @media #{$desktopUp} {
      padding: 2em;
    }
  }

  .headerLink {
    color: #fff;
    display: block;
    font-family: sans-serif;
    font-size: 1.5em;
    margin: 0 0 0.5em;
    text-align: center;
    text-decoration: none;

    @media #{$tabletUp} {
      margin: 0;
    }
  }

  .search {
    margin: 0 0 0 auto;

    @media #{$tabletUp} {
      width: calc(50% - 1em);
    }

    input {
      padding: 1em;
      width: 100%;
    }
  }
}
