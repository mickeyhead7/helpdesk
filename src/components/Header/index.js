import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Link, { navigateTo } from 'gatsby-link';

import './styles.scss';

class Header extends Component {
  constructor(props) {
    super();

    this.state = {
      searchValue: props.search,
    };
  }

  handleChange = (event) => {
    this.setState({
      searchValue: event.target.value,
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();

    navigateTo(`/search?search=${this.state.searchValue}`);
  }

  render() {
    return (
      <div className="header">
        <div className="headerContainer">
          <header>
            <Link
              to="/"
              className="headerLink"
            >
              Helpdesk
            </Link>
          </header>
          <form
            className="search"
            onSubmit={this.handleSubmit}
          >
            <label htmlFor="search">
              <input
                defaultValue={this.state.searchValue}
                id="search"
                onChange={this.handleChange}
                placeholder="Search"
                type="text"
              />
            </label>
          </form>
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  search: PropTypes.string,
};

Header.defaultProps = {
  search: '',
};

export default Header;
