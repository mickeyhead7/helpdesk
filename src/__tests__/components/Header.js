import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../../components/Header';

const search = 'search text';

test('Header', () => {
  const component = mount(<Header search={search} />);

  expect(component.state('searchValue')).toEqual(search);
  expect(toJson(component)).toMatchSnapshot();
});

test('Entering a search string should update the component state', () => {
  const component = mount(<Header search={search} />);
  const searchInput = component.find('input#search');
  const newValue = 'New search text';

  searchInput.simulate('change', {
    target: {
      value: newValue,
    },
  });

  expect(component.state('searchValue')).toEqual(newValue);
});
