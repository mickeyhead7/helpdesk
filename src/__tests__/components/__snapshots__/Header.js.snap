// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
<Header
  search="search text"
>
  <div
    className="header"
  >
    <div
      className="headerContainer"
    >
      <header>
        <Link
          className="headerLink"
          to="/"
        >
          <a
            href="/"
          >
            Helpdesk
          </a>
        </Link>
      </header>
      <form
        className="search"
        onSubmit={[Function]}
      >
        <label
          htmlFor="search"
        >
          <input
            defaultValue="search text"
            id="search"
            onChange={[Function]}
            placeholder="Search"
            type="text"
          />
        </label>
      </form>
    </div>
  </div>
</Header>
`;
